app/javascript/components/EditForm.js

```diff
@@ -76,33 +94,43 @@ class EditRebateForm extends React.Component {
             <div className="flex-row">
               {map(customerDetailFields, (field) => {
                 return field.type == 'radio'
-                ? RadioInput(field)
-                : SingleInput(field)
+                ? RadioInput({...field, isEditable})
+                : SingleInput({...field, isEditable})
               })}
             </div>
             { values.fields.moved_within_rating_year == 'yes'
             ? <div className="flex-row">
                 {map(conditionalsFields, (field) => {
                 return field.type == 'radio'
-                ? RadioInput(field)
-                : SingleInput(field)
+                ? RadioInput({...field, isEditable})
+                : SingleInput({...field, isEditable})
                 })}
               </div>
               : null
             }
-            {IncomeDeclaration()}
-            <div className="buttons">
+            {IncomeDeclaration({otherIncomeFields, isEditable})}
+            <div className={'flex-row'}>
+              <Field
+                className='rebate-search-input flex-item'
+                name="newIncomeField"
+                component="input"
+                readOnly={!isEditable}
+              />
               <button
+                className='one-third rebate-add-income-button'
+                disabled={!isEditable || !(values.newIncomeField)}
```

Comment:
if this button should be enabled if it is `editable` *and* it is for a new incomefield, then I think writing it like
`! (isEditable && values.newIncomeField)` is easier to understand

Reply:
having made changed `isEditable` to `isReadOnly` I find
`isReadOnly || !values.newIncomeField`
clearer then
`!(!isReadOnly && values.newIncomeField)`